File: cs50x/pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text, int length); 
int count_words(string text, int length);
int count_sentences(string text, int length);
int main(void)
{
    string text = get_string("Text: ");
    int length = strlen(text);
    int letters = count_letters(text, length);
    int words = count_words(text, length);
    int sentences = count_sentences(text, length);
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int count_letters(string text, int length)
{

    int letters = 0;
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text, int length)
{
    int words = 1;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text, int length)
{
    int sentences = 0;
    for (int i = 0; i < length; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: cs50x/pset6/sentimental-readability/readability.py
from math import *


def main():
    text = input("Text: ")
    length = len(text)
    letters = count_letters(text, length)
    words = count_words(text, length)
    sentences = count_sentences(text, length)
    L = float(letters / words * 100)
    S = float(sentences / words * 100)
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)

    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text, length):
    letters = 0
    for i in range(length):
        if text[i].isalpha():
            letters += 1
    return letters


def count_words(text, length):
    words = 1
    for i in range(length):
        if text[i] == ' ':
            words += 1
    return words


def count_sentences(text, length):
    sentences = 0
    for i in range(length):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sentences += 1
    return sentences


main()
